### PR TITLE
Populate 'Files changed' sidebar section from tool call events (#207)

### PR DESCRIPTION
**Pull request by pswitchy**
The "Files changed" section in the session right sidebar was previously non-functional because it always received an empty array. This PR implements a client-side utility to extract and compute file changes directly from [SandboxEvent](cci:2://file:///e:/background-agents/packages/web/src/components/session-right-sidebar.tsx:35:0-40:1) tool calls (`Edit` and [Write](cci:2://file:///e:/background-agents/packages/web/src/lib/tasks.ts:13:0-19:1)).

## Key Changes
- **New Utility**: Created [extractChangedFiles(events)](cci:1://file:///e:/background-agents/packages/web/src/lib/files.ts:23:0-68:1) in [packages/web/src/lib/files.ts](cci:7://file:///e:/background-agents/packages/web/src/lib/files.ts:0:0-0:0).
  - Filters for completed `Edit` and [Write](cci:2://file:///e:/background-agents/packages/web/src/lib/tasks.ts:13:0-19:1) tool calls.
  - Estimates additions/deletions based on line counts of `newString`/`oldString` (Edit) or `content` (Write).
  - Deduplicates by filename and accumulates stats across multiple operations.
  - Sorts alphabetically for stable display.
- **Sidebar Integration**: Modified [packages/web/src/components/session-right-sidebar.tsx](cci:7://file:///e:/background-agents/packages/web/src/components/session-right-sidebar.tsx:0:0-0:0).
  - Removed the unused `filesChanged` prop.
  - Sidebar now derives `filesChanged` internally using `useMemo` and the new extraction utility.
  - This automatically enables file change tracking across desktop, tablet, and mobile views.

## Verification
- **Automated Tests**: Added 15 comprehensive test cases in [packages/web/src/lib/files.test.ts](cci:7://file:///e:/background-agents/packages/web/src/lib/files.test.ts:0:0-0:0) covering path resolution, stat estimation, deduplication, and edge cases.
- **Test Results**: All 49 tests passed (including existing tests).
  ```bash
  Test Files  4 passed (4)
       Tests  49 passed (49)

### CODE REVIEW

File: packages/web/next-env.d.ts [hunk 1]
```diff
@@ -1,6 +0,0 @@
-/// <reference types="next" />
```
Comment (ColeMurray) [inline packages/web/next-env.d.ts:1]: [Automated Review] **Unrelated change** — This auto-generated Next.js type declaration file is referenced in `tsconfig.json` (`"include": ["next-env.d.ts", ...]`). Next.js regenerates it on `next dev`/`next build`, so builds won't break, but deleting it from version control is unrelated to this PR and will cause the file to reappear as untracked after the next build.

This deletion should be reverted — or if intentional, it belongs in a separate commit with `next-env.d.ts` added to `.gitignore`.

File: packages/web/src/lib/files.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+import type { FileChange } from "@/types/session";
+
+interface SandboxEvent {
+  type: string;
+  tool?: string;
+  args?: Record<string, unknown>;
+  status?: string;
+  timestamp: number;
+}
```
Comment (ColeMurray) [inline packages/web/src/lib/files.ts:14]: [Automated Review] **Type safety gap** — This introduces a 5th local copy of `SandboxEvent` across the web package. Critically, the sidebar's `SandboxEvent` (`session-right-sidebar.tsx:37-42`) **lacks the `status` field** that `extractChangedFiles` depends on for filtering (`event.status !== "completed"` at line 35).

This works at runtime because the actual objects from `useSessionSocket` carry `status`, but the type contract doesn't guarantee it. If someone ever refactors to pass only the typed fields, `extractChangedFiles` would silently return an empty array.

Consider either:
- Adding `status?: string` to the sidebar's `SandboxEvent`, or
- Consolidating `SandboxEvent` into a single shared type (e.g., in `@/types/session` alongside `FileChange`)

File: packages/web/src/lib/files.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    let deletions = 0;
+
+    if (normalizedTool === "edit") {
+      additions = countLines(args.newString);
+      deletions = countLines(args.oldString);
+    } else {
+      // write
+      additions = countLines(args.content);
+      deletions = 0;
+    }
```
Comment (ColeMurray) [inline packages/web/src/lib/files.ts:85]: [Automated Review] **Inaccurate diff stats** — `countLines` counts *total* lines in `newString`/`oldString`, not the actual changed lines. For example, replacing `"a\nb"` → `"a\nb\nc"` reports +3/−2 when the real change is +1/−0. This inflates stats significantly for large edits where most lines are unchanged.

**Recommendation:** Use `diffLines` from the [`diff`](https://www.npmjs.com/package/diff) package (~3.2 kB gzipped, tree-shaken; built-in TypeScript types since v8). It returns `Change[]` objects with `.added`, `.removed`, and `.count` fields, giving accurate line-level addition/deletion counts:

```typescript
import { diffLines } from 'diff';

function computeEditStats(oldStr: string, newStr: string): { additions: number; deletions: number } {
  const changes = diffLines(oldStr, newStr);
  let additions = 0;
  let deletions = 0;
  for (const change of changes) {
    if (change.added) additions += change.count;
    if (change.removed) deletions += change.count;
  }
  return { additions, deletions };
}
```

For Write operations (no prior content to diff against), the current `countLines(content)` approach is fine.

File: packages/web/next-env.d.ts (hunk 1 above)
Comment (ColeMurray) [reply packages/web/next-env.d.ts:1]: Done — restored `next-env.d.ts` from `main`. This deletion was unintentional and unrelated to the PR. Fixed in 6fcfec4.

File: packages/web/src/lib/files.ts (hunk 3 above)
Comment (ColeMurray) [reply packages/web/src/lib/files.ts:85]: Done — replaced the `countLines` heuristic with `diffLines` from the `diff` package (v8, ~3.2 kB gzipped). Edit events now use accurate line-level diffing via `computeEditStats`, with `ignoreNewlineAtEof: true` to handle tool call substrings cleanly. Write events still use `countLines` (no prior content to diff against).

Added a test case demonstrating the improvement: a 5-line edit with 1 changed line correctly reports +1/−1 instead of the old +5/−5.

Fixed in 9fbe48b.

File: packages/web/src/lib/files.ts (hunk 2 above)
Comment (ColeMurray) [reply packages/web/src/lib/files.ts:14]: Acknowledged — the type safety gap is real. The sidebar's `SandboxEvent` lacks `status`, so `extractChangedFiles` relies on runtime objects carrying it even though the type contract doesn't guarantee it.

The minimal fix is adding `status?: string` to the sidebar's `SandboxEvent`. Full consolidation of all 5 `SandboxEvent` copies into a shared type (e.g., in `@/types/session`) would be the proper long-term fix but is out of scope for this PR.
